presto-spi/src/main/java/com/facebook/presto/spi/block/DictionaryBlock.java

```diff
@@ -321,7 +321,7 @@ public Block copyPositions(int[] positions, int offset, int length)
     public Block getRegion(int positionOffset, int length)
     {
         checkValidRegion(positionCount, positionOffset, length);
-        return new DictionaryBlock(idsOffset + positionOffset, length, dictionary, ids, false, randomDictionaryId());
+        return new DictionaryBlock(idsOffset + positionOffset, length, getDictionary(), ids, false, getDictionarySourceId());
```

Comment:
What was the bug (cmt msg doesn't suggest anything in particular) and whether it would be possible to test this?

Reply:
A description of the bug is featured prominently and described in some detail in #11727.  The bug itself is end-to-end in nature and I'm not sure of any greater coverage that could be included that was already included in #11727, but I'm open to suggestions.  Thank you for pointing out that the commit message isn't completely adequate in describing the nature of the change, let me see what I can do to fix that.